components/script/dom/gpucommandencoder.rs

```diff
@@ -68,18 +92,52 @@ impl GPUCommandEncoderMethods for GPUCommandEncoder {
         &self,
         _descriptor: &GPUComputePassDescriptor,
     ) -> DomRoot<GPUComputePassEncoder> {
-        GPUComputePassEncoder::new(&self.global(), self.channel.clone(), self.encoder)
+        *self.state.borrow_mut() = GPUCommandEncoderState::EncodingComputePass;
```

Comment:
We'd need to have a state check here (i.e. it has to be `Open`).
We could pass down the expected state to `set_state` and then turn it into invalid if the expectation doesn't match.
Not sure if you want to do it here on in the follow-up

Reply:
I'll do it here itself.